datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -505,7 +507,35 @@ pub async fn from_substrait_rel(
                     _ => Ok(t),
                 }
             }
-            _ => not_impl_err!("Only NamedTable reads are supported"),
+            Some(ReadType::VirtualTable(vt)) => {
+                let base_schema = read
+                    .base_schema
+                    .clone()
+                    .expect("expected schema to exist for virtual table");
+
+                let fields = from_substrait_named_struct(&base_schema)?;
+                let schema = DFSchemaRef::new(DFSchema::try_from(Schema::new(fields))?);
```

Comment:
I suspect that the schema here may not be necessary, perhaps it can be restored through [LogicalPlanBuilder::values](https://github.com/apache/datafusion/blob/f80794779ef251528ee983e22ec03c158deb42a6/datafusion/expr/src/logical_plan/builder.rs#L168).

Reply:
It probably could, except for the column names - LogicalPlanBuilder::values names everything to just `column1` etc. Do you see some benefit to using that (and then handling the column names somehow) instead of passing in the schema explicitly?

Reply:
I once wanted to remove the schema, but we need to use it to pass the struct's names, so the current approach should be fine.